Extension/src/Utility/Process/program.ts

```diff
@@ -0,0 +1,172 @@
+            }
+
+            // we were given a valid executable path, so we'll just check with the event listeners real quick.
+            const bin = await emitNow<string>('select-binary', Descriptors.none, executable, is.promise(opts.choices) ? await opts.choices : new Set()) || executable;
+
+            // ensure that the executable is an absolute path
+            asserts.isAbsolute(bin);
+            // ensure that the executable exists and is executable
+            await asserts.isExecutable(bin);
+            return bin;
+        });
+    } else {
+        cmdlineArgs = [...(executable as Launcher).cmdlineArgs, ...cmdlineArgs];
+        opts = { ...opts, ...((executable as Launcher).options || {}) };
+        fullPath = (executable as Launcher).executable;
+    }
+
+    let result: ProcessFunction | CommandFunction;
+    if (opts.noninteractive) {
+        // create launcher for non-interactive commands
+        result = (async (...moreArgs: ArrayPlusOptions<Primitive, ProgramOptions>): Promise<CommandResult> => {
+            // make sure the path is executable
+
+            // Create Process instance
+            const moreOpts = options(moreArgs);
+            const subscribers = opts.on ? moreOpts.on ? [opts.on, moreOpts.on] : [opts.on] : moreOpts.on ? [moreOpts.on] : [];
+
+            const proc = await new Process(
+                await fullPath,                                                                                   // executable
+                [...result.cmdlineArgs, ...primitives(moreArgs)],                                                  // arguments
+                result.options.cwd || moreOpts.cwd || process.cwd(),                                              // cwd
+                { ...process.env, ...result.options.env, ...moreOpts.env },                                       // environment
+                false,                                                                                            // no stdin.
+                ...subscribers                                                                                    // event handlers
+            );
+
+            const code = await proc.exitCode;
+            return {
+                code,
+                stdio: proc.stdio,
+                error: proc.error
+            };
+
+        }) as any as CommandFunction;
+    } else {
+        // create Launcher function
+        result = (async (...moreArgs: ArrayPlusOptions<Primitive, ProgramOptions>): Promise<Process> => {
+            // Create Process instance
+            const moreOpts = options(moreArgs);
+            const subscribers = opts.on ? moreOpts.on ? [opts.on, moreOpts.on] : [opts.on] : moreOpts.on ? [moreOpts.on] : [];
+            return new Process(
+                await fullPath,                                                                                   // executable

```

Comment:
The large amount of space before the `//` may cause the comment to be hidden off the screen unnecessarily.

Reply:
reducing whitespace to 1

Reply:
added an eslint rule to eliminate extra spacing